File: xr.r
# Initialize data
real_data <- seq(1, 10, by = 1)  # This will explicitly set real_data as numeric
int_data <- 1:10                 # Integer data

# Format function for aligned output with colon
fmt_real <- function(label, values) {
  cat(sprintf("%32s: ", label))  # Add the colon and space after label
  cat(sprintf("%10.4f", values), "\n")
}

fmt_int <- function(label, values) {
  cat(sprintf("%32s: ", label))  # Add the colon and space after label
  cat(sprintf("%10d", values), "\n")
}

fmt_real("Real data", real_data)
fmt_int("Integer data", int_data)
fmt_real("Mean of real_data", mean(real_data))
fmt_real("Mean of int_data", mean(int_data))
fmt_real("Standard deviation of real_data", sd(real_data))
fmt_real("Standard deviation of int_data", sd(int_data))
fmt_real("Variance of real_data", var(real_data))
fmt_real("Variance of int_data", var(int_data))
fmt_real("Median of real_data", median(real_data))
fmt_real("Median of int_data", median(int_data))
fmt_real("Range of real_data", range(real_data))
fmt_int("Range of int_data", range(int_data))
quantiles_real <- quantile(real_data, probs = c(0.25, 0.50, 0.75))
quantiles_int <- quantile(int_data, probs = c(0.25, 0.50, 0.75))
fmt_real("Quantiles of real_data", quantiles_real)
fmt_real("Quantiles of int_data", quantiles_int)
fmt_real("IQR of real_data", IQR(real_data))
fmt_real("IQR of int_data", IQR(int_data))
fmt_real("Product of real_data", prod(real_data))
fmt_int("Product of int_data", prod(int_data))

File: xr.f90
! Description: Main program to demonstrate statistical functions from r_mod
program StatisticsExample
    use r_mod
    implicit none
    character (len=*), parameter :: fmt_cr = "(A30, *(1x,f10.4))", fmt_ci = "(A30, *(1x,i10))"
    integer, parameter :: n = 10
    integer :: i
    real(kind=dp) :: real_data(n)
    integer :: int_data(n)
    int_data = [(i, i=1, n)]
    real_data = real(int_data, kind=dp)
    write(*, fmt_cr) "Real data:", real_data
    write(*, fmt_ci) "Integer data:", int_data
    write(*, fmt_cr) "Mean of real_data:", mean(real_data)
    write(*, fmt_cr) "Mean of int_data:", mean(int_data)
    write(*, fmt_cr) "Standard deviation of real_data:", sd(real_data)
    write(*, fmt_cr) "Standard deviation of int_data:", sd(int_data)
    write(*, fmt_cr) "Variance of real_data:", var(real_data)
    write(*, fmt_cr) "Variance of int_data:", var(int_data)
    write(*, fmt_cr) "Median of real_data:", median(real_data)
    write(*, fmt_cr) "Median of int_data:", median(int_data)
    write(*, fmt_cr) "Range of real_data:", min_max(real_data)
    write(*, fmt_ci) "Range of int_data:", min_max(int_data)
    write(*, fmt_cr) "Quantiles of real_data:", quantile(real_data, [0.25_dp, 0.5_dp, 0.75_dp])
    write(*, fmt_cr) "Quantiles of int_data:", quantile(int_data, [0.25_dp, 0.5_dp, 0.75_dp])
    write(*, fmt_cr) "IQR of real_data:", IQR(real_data)
    write(*, fmt_cr) "IQR of int_data:", IQR(int_data)
    write(*, "(A30, *(1x,f14.4))") "Product of real_data:", prod(real_data)
    write(*, fmt_ci) "Product of int_data:", prod(int_data)
end program StatisticsExample
